jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/LocalBaseImageSteps.java

```diff
@@ -90,25 +91,58 @@ static boolean isGzipped(Path path) throws IOException {
 
   private final ExecutorService executorService;
   private final BuildConfiguration buildConfiguration;
-  private final Path tarPath;
   private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
-  private final TempDirectoryProvider tempDirectoryProvider;
 
-  ExtractTarStep(
+  LocalBaseImageSteps(
       ExecutorService executorService,
       BuildConfiguration buildConfiguration,
-      Path tarPath,
-      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
-      TempDirectoryProvider tempDirectoryProvider) {
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory) {
     this.executorService = executorService;
     this.buildConfiguration = buildConfiguration;
-    this.tarPath = tarPath;
     this.progressEventDispatcherFactory = progressEventDispatcherFactory;
-    this.tempDirectoryProvider = tempDirectoryProvider;
   }
 
-  @Override
-  public LocalImage call()
+  Callable<LocalImage> processDockerDaemonBaseImageStep(DockerClient dockerClient) {
+    return () -> {
+      ImageReference imageReference = buildConfiguration.getBaseImageConfiguration().getImage();
+      try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider();
+          ProgressEventDispatcher progressEventDispatcher =
+              progressEventDispatcherFactory.create("processing base image " + imageReference, 2)) {
+        Path outputDir = tempDirectoryProvider.newDirectory();
+        Path tarPath = outputDir.resolve("out.tar");
+        try (TimerEventDispatcher ignored =
+            new TimerEventDispatcher(
+                buildConfiguration.getEventHandlers(),
+                "Saving " + imageReference + " from Docker daemon")) {
+          long size = dockerClient.inspect(imageReference).getSize();
+          try (ProgressEventDispatcher dockerProgress =
+                  progressEventDispatcher
+                      .newChildProducer()
+                      .create("saving base image " + imageReference, size);
+              ThrottledAccumulatingConsumer throttledProgressReporter =
+                  new ThrottledAccumulatingConsumer(dockerProgress::dispatchProgress)) {
+            dockerClient.save(imageReference, tarPath, throttledProgressReporter);
+          }
+        }
+
+        return extractTar(
+            tarPath, tempDirectoryProvider, progressEventDispatcher.newChildProducer());
+      }
+    };
+  }
+
+  Callable<LocalImage> processTarBaseImageStep(Path tarPath) {
+    return () -> {
+      try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
+        return extractTar(tarPath, tempDirectoryProvider, progressEventDispatcherFactory);
+      }
+    };
+  }
+
+  private LocalImage extractTar(
```

Comment:
Suggest rename to `handleDockerImageTar()`?  Since this does more than extract a tar file — and it's `docker-save` specific (not OCI images, for example).